docs/module_web_with_python.md

```diff
@@ -5,137 +5,437 @@
 ### Clean code
 
 #### Point out 5 suggestions, how to format an SQL query!
+* Separate your query into multiple lines.
+* Capitalize keywords such as "WHERE, HAVING, SELECT" for better readability.
+* Use "AS" to rename columns when needed for a prettier display.
+* Use underscore when naming tables/columns with multiple words in them.
+* Avoid giving the same name for both the table and a column.
 #### What layers can you name in a simple web application?
-
+* Front-end (What happens on the user's computer.)
+* Back-end (What happens on the server's side.)
+* Data storage and access (Usually SQL Datbase, or CSV file.)
 ### Error handling
 #### What error can occur, when an array does not have an element on the requested index?
+* If the requested index is bigger than the length of the array, then an index error occurs.
+* Else, the program runs as it would normally, however, the returned value will be empty.
 #### What is the “finally” block, and how would you use it?
+* The finally block is part of a try-except block.
+* A try-except block is used to catch special exceptions types in a program.
+* The finally block runs once the try-except block has finished it's tasks. 
 #### Why should we catch special exception types?
-
+* Exceptions are prone to break the running of our code when they occur.
+* Catching exceptions allows us to do a workaround when an exception occurs so that they won't break our code.
 ### Security
 #### What is SQL injection? How to protect an application against it?
+* SQL injection is a code injection technique in which a client inserts malicious SQL statements into an entry field for execution.
+* To protect your applications from SQL Injection. Check all data coming from the user and use scripts or libraries to catch keywords in the data.
+* example: Hello, my name is'); DROP DATABASE users;
 #### What is XSS? How to protect an application against it?
+* XSS, Cross-Site-Scripting is a security breach which takes advantage of dynamically generated webpages.
+* An XSS attack happens when a client injects malicious third party scripts into a dynamically generated website.
+* Most common script types include: Javascript, HTML, Flash <- now obsolete.
+* Via XSS, the client is able to: highjack accounts, change user settings, and/or poison cookies with malicious code.
 #### How to properly store passwords?
+* Password should be stored after put through a hashing (encoding) algorithm for increased security.
```

Comment:
Might worth to note that salting them after hashing is also recommended.

Reply:
fixed